.cbmc-batch/source/openssl/ec_override.c

```diff
@@ -0,0 +1,443 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <limits.h>
+
+#include <ec_utils.h>
+#include <openssl/ec.h>
+#include <openssl/objects.h>
+#include <bn_utils.h>
+
+#include <proof_helpers/nondet.h>
+#include <proof_helpers/proof_allocators.h>
+
+/* Abstraction of the EC_GROUP struct */
+struct ec_group_st {
+    int curve_name;
+    point_conversion_form_t asn1_form;
+  BIGNUM* order;
+};
+
+/*
+ * Description: In order to construct a builtin curve use the function EC_GROUP_new_by_curve_name and provide the nid of
+ * the curve to be constructed. Return values: All EC_GROUP_new* functions return a pointer to the newly constructed
+ * group, or NULL on error.
+ */
+EC_GROUP *EC_GROUP_new_by_curve_name(int nid) {
+    assert(nid == NID_X9_62_prime256v1 || nid == NID_secp384r1);
+
+    EC_GROUP *group = can_fail_malloc(sizeof(EC_GROUP));
+
+    if (group) {
+        group->curve_name = nid;
+        group->asn1_form  = POINT_CONVERSION_UNCOMPRESSED;
+	group->order = bignum_nondet_alloc();
+	__CPROVER_assume(bignum_is_valid(group->order)); // Can we assume order is always initialized?
```

Comment:
I believe this is a safe assumption. If this fails the function should clean up and return null, so this failure case would be indistinguishable from a null return at line 41.